app/models/import_users.rb

```diff
@@ -0,0 +1,81 @@
+class ImportUsers < Struct.new(:import, :content_path)
+  def perform
+    content_hash = JSON.parse(File.read(content_path), :symbolize_names => true)
+    
+    duplicate_users = []
+    user_for_lara = []
+    total_users_count = content_hash[:users].size
+
+    import.update_attribute(:total_imports, total_users_count)
+
+    content_hash[:users].each_with_index do |user, index|
+      new_user = User.find_by_login(user[:login]) || User.find_by_login(user[:email])
+      unless new_user
+        new_user = User.create({
+          :email => user[:email],
+          :login => user[:login],
+          :last_name => user[:last_name],
+          :first_name => user[:first_name],
+          :password => "password", #default password
+          :password_confirmation => "password",
+          :require_password_reset => true
+          
+        }){|u| u.skip_notifications = true}
+        new_user.confirm!
+        user[:roles].each do |role|
+          new_user.add_role(role)
+        end
+        if user[:teacher]
+          if user[:school]
+            if user[:school][:district]
+              district_params = {}
+              district_params[:name] = user[:school][:district][:name]
+              district_params[:leaid] = user[:school][:district][:leaid]
+              district = Portal::District.find(:first, :conditions => district_params)
```

Comment:
This should use the district state if it has one.
Also if the imported user's district leaid is null, then it would still be ok to put them in an existing district in the same state with the same name. Regardless of whether that existing district has an leaid or not. The current code above  in these conditions would only match districts with an null leaid.